### PR TITLE
Update composer.json dependencies

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Ensures dependencies are downloaded when using composer.

# Needed By (Date)
- ?

# Urgency
- Semi-urgent. 

# Steps to Test

1. Change all of the dependencies to stanford_courses in acsf.cardinald7 to this branch
2. Run composer update
3. Validate that the new modules have downloaded
4. Enable the course_views module if not enabled
5. Re-install a product site like jsplus or jsa

# Affected Projects or Products
- Stanford Courses
- Most of our products

# Associated Issues and/or People
- SITES-903

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_course_views/stanford_course_views.info [hunk 1]
```diff
@@ -9,7 +9,7 @@ dependencies[] = stanford_courses
 dependencies[] = stanford_image
 dependencies[] = views
 dependencies[] = views_bulk_operations
-dependencies[] = views_filters_selective
+dependencies[] = views_selective_filters
```
Comment (sherakama) [inline modules/stanford_course_views/stanford_course_views.info]: This might be wrong. Not sure what the module is called on dl. Should check.

Comment (jbickar) [reply modules/stanford_course_views/stanford_course_views.info]:
```
drush pmi views_filters_selective
 Extension        :  views_filters_selective                                                                           
 Project          :  views_selective_filters                                                                           
 Type             :  module                                                                                            
 Title            :  Views Selective Exposed Filters                                                                   
 Description      :  Restrict exposed filter values to those present in the result set.                                
 Version          :  7.x-1.5                                                                                           
 Date             :  2016-01-01                                                                                        
 Package          :  Views                                                                                             
 Core             :  7.x                                                                                               
 PHP              :  5.2.4                                                                                             
 Status           :  enabled                                                                                           
 Path             :  sites/default/modules/contrib/views_selective_filters                                             
 Schema version   :  module has no schema                                                                              
 Files            :  views_handler_filter_selective.inc                                                                
 Requires         :  views (>=7.x-3.11), ctools                                                                        
 Required by      :  stanford_course_views, stanford_courses_person_reference_views, stanford_sites_jumpstart_academic 
 Permissions      :  none                                                                                              
 Configure        :  None   
```
The project name is views_selective_filters. The module name is views_filters_selective.

Comment (sherakama) [reply modules/stanford_course_views/stanford_course_views.info]: Danke.

Comment (jbickar) [reply modules/stanford_course_views/stanford_course_views.info]: So this should be `views_filters_selective` in the .info file, ja?

Comment (sherakama) [reply modules/stanford_course_views/stanford_course_views.info]: Jah, .info files need it to be the module name as it lies in the file system. Composer needs it to be the name of the Drupal project as it shows in the url.

Comment (jbickar) [reply modules/stanford_course_views/stanford_course_views.info]: So this change should be reverted...

Comment (sherakama) [reply modules/stanford_course_views/stanford_course_views.info]: It has....

Comment (jbickar) [reply modules/stanford_course_views/stanford_course_views.info]: That's what I get for commenting on an outdated commit 🤦‍♂️

**Comment by jbickar** [comment]
Was Last Week John blind? https://github.com/SU-SWS/acsf-cardinald7/tree/1.x/docroot/sites/all/modules/contrib/views_selective_filters

**Comment by jbickar** [comment]
```
spr-jbickar-4: ~/Documents/working/ACSF/acsf-leland$ find . -name composer.json | xargs grep -n "drupal/features"
./docroot/sites/all/modules/stanford/stanford_courses/composer.json:11:      "drupal/features": "~2.0",
./docroot/profiles/stanford/composer.json:318:      "drupal/features": "~2.10",
./docroot/profiles/stanford/composer.json:441:      "drupal/features": {
./composer.json:140:      "drupal/features": {
./vendor/drush/drush/tests/makefiles/composer.json:16:        "drupal/features": "7.1.0-beta4",
./vendor/drush/drush/tests/makefiles/composer.json:44:            "drupal/features": {
./vendor/drush/drush/tests/makefiles/composer.json:45:                "Enter drupal/features patch #0 description here": "http://drupal.org/files/issues/features-drush-backend-invoke-25.patch"
```

**Comment by sherakama** [comment]
This is a gap between the original Drupal packager and the new composer based workflow.
Some of the modules do not get their version in the .info file and it is up to (I Think) `composer_autoloader` to patch this up for Drupal.

**Comment by jbickar** [comment]
_WAT DO_

**Comment by sherakama** [comment]
Check that `composer_autoloader` is enabled and if not enable it first.

**Comment by sherakama** [comment]
failing that, try updating the stack with `composer update`

**Comment by jbickar** [comment]
🤔

**Comment by jbickar** [comment]
`composer_autoloader` no worky. Also not able to enable through the GUI.

**Comment by jbickar** [comment]
I'm trying `composer update` now

**Comment by jbickar** [comment]
I assume `stanford_course_views` is something that we want to be available to self-service users? Trying to determine how far down this rabbit hole I should go...

**Comment by sherakama** [comment]
https://github.com/SU-SWS/acsf-leland/blob/1.x/docroot/sites/all/modules/contrib/features/features.info is missing the packager information but https://github.com/SU-SWS/acsf-cardinald7/blob/1.x/docroot/sites/all/modules/contrib/features/features.info is not.

**Comment by sherakama** [comment]
Remove it and re-add it. Composer.lock probably thinks it is on 2.10 and is not going to change the files.

**Comment by jbickar** [comment]
👌

**Comment by jbickar** [comment]
My sweater is slowly unraveling because I keep pulling on this thread. https://stanfordits.atlassian.net/browse/SITES-909

**Comment by jbickar** [comment]
`stanford_feeds_helper` is a dependency of `stanford_courses`. Should that be in this PR?

I will review for other submodule dependencies. I can finally enable `stanford_course_views` with the `leland` codebase.

**Comment by jbickar** [comment]
## Undeclared Submodule Dependencies
- `stanford_feeds_helper`
- `stanford_image`
- `ds`
- `entityreference`
- `context`
- `stanford_person`

**Comment by jbickar** [comment]
Per discussion, I'll add the three missing Drupal contrib modules to `composer.json`, but skip the Stanford ones for now, as they are not on Packagist, and we have concerns about performance if we add them as VCS repos.

**Pull request merged**